### PR TITLE
fixing non contiv node vm scenario

### PR DESCRIPTION
**Pull request by abhi**

### CODE REVIEW

**Comment by mapuri** [comment]
@abhinandanpb for non vagrant vms you should use `Baremetal` Testsuite. `Vagrant` Testsuite is specially meant  for vagrant setups as it takes over the bringup and teardown of the setup as well.

For one way of using `Baremetal` testsuite take a look at  cluster repo:

https://github.com/contiv/cluster/blob/master/management/src/Makefile#L97

https://github.com/contiv/cluster/blob/master/management/src/systemtests/cli_test.go#L52

File: vagrant.go [hunk 1]
```diff
@@ -135,11 +135,6 @@ func (v *Vagrant) setup(start bool, env string, numNodes int) error {
 		nodeName := strings.Fields(string(nodeNameByte))[0]
 		nodeNames = append(nodeNames, nodeName)
 	}
-	if len(nodeNames) != numNodes {
```
Comment (mapuri) [inline vagrant.go:138]: not doing this check will break the case when some node fail to come up, right?

**Comment by abhi** [comment]
@mapuri sorry for the wrong title descrition. Corrected it now.

File: vagrant.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func (v *Vagrant) setup(start bool, env string, numNodes int) error {
 		}()
 	}
 
-	v.expectedNodes = numNodes
+	v.expectedNodes = numContivNodes+numIgnoreNodes
```
Comment (mapuri) [inline vagrant.go]: why not just pass numNodes as the sum of the two from the caller?

**Comment by mapuri** [comment]
@abhinandanpb 

I have a few concerns adding `numIgnoredNodes` to the `Setup()` for a vagrant testbed type. This testbed type is really meant for environment where the **TestbedNode**s are brought up using the `testbed.Setup()`, so adding an ignored node count seems to overload the `Vagrant.Setup()`.

We can try to mold this function to fit our current needs but it will make this testbed type more specific to our environments, than a generic Vagrant based test setup completely managed by `Setup()` and `TearDown()` functions.

I still think the Baremetal testbed type matches your scenario better. `Baremetal` might be a misnomer but it provides what you are looking for i.e. it expects a bunch of nodes to be already up and then user decides to add all or subset of them to the testbed nodes themselves.

Let's discuss more today and see if we can address you use case more cleanly.

File: vagrant.go [hunk 3]
```diff
@@ -214,7 +209,11 @@ func (v *Vagrant) Setup(args ...interface{}) error {
 	if _, ok := args[2].(int); !ok {
 		return unexpectedSetupArgError("bool, string, int", args...)
 	}
-	return v.setup(args[0].(bool), args[1].(string), args[2].(int))
+        if _, ok := args[3].(int); !ok {
+                return unexpectedSetupArgError("bool, string, int", args...)
```
Comment (mapuri) [inline vagrant.go]: s/ "bool, string, int" / "bool, string, int, int"

File: vagrant.go (hunk 2 above)
Comment (abhi) [inline vagrant.go]: Hmm ok so here is the problem . The code in vagrantssh assumes num nodes to be contiv_nodes only. Hence the ignore nodes.

**Comment by abhi** [comment]
@mapuri the code assumes this to be used with contiv only from what I see. We can discuss this offline

**Comment by mapuri** [comment]
@abhinandanpb LGTM, thanks for incorporating the comments. This solution is a lot cleaner imo.

Looks like the current sanity run is going to take a bit as it is running on bldg14 machines where box download is taking a lot longer. If you want you can resubmit the patch which will schedule the next job on bldg1 machines which should be faster.

**Comment by abhi** [comment]
@mapuri i have resubmitted it. Thanks !

**Pull request merged**